src/Microsoft.ML.Transforms/LambdaTransform.cs

```diff
@@ -68,9 +68,13 @@ private static ITransformer Create(IHostEnvironment env, ModelLoadContext ctx)
 
             var contractName = ctx.LoadString();
 
-            var composition = env.GetCompositionContainer();
-            ITransformer transformer = composition.GetExportedValue<ITransformer>(contractName);
-            return transformer;
+            object factoryObject = env.ComponentCatalog.GetExtensionValue(env, typeof(CustomMappingTransformerFactoryAttribute), contractName);
+            if (!(factoryObject is ICustomMappingTransformerFactory transformerFactory))
+            {
+                throw env.Except($"The class with contract '{contractName}' must derive from '{typeof(CustomMappingTransformerFactory<,>).FullName}'.");
```

Comment:
>CustomMappingTransformerFactory [](start = 102, length = 31)

Custom mappings, at least historically within Microsoft, have been *very* popular, so eventually we'll want to probably make this sort of rule part of our code analyzer. (Obviously not right now! Just something I'd like to think about for post v1.)

Reply:
Ha ha, and considering the exclusion rules, the internal code analyzer would have to make sure that it is *never* used. :)

---
In reply to: [257425601](https://github.com/dotnet/machinelearning/pull/2569#discussion_r257425601) [](ancestors = 257425601)

Reply:
Logged #2670 to address this.